### PR TITLE
Handle cable pulled scenario on wspp websocket stack

### PR DESCRIPTION
**Pull request by jplafonta**

### CODE REVIEW

File: Source/Common/utils.cpp [hunk 1]
```diff
@@ -76,6 +76,47 @@ bool StringToUint(String const& s, uint64_t& v, int32_t base)
...
 }
 
+HRESULT RunAsync(
+    AsyncWork&& work,
+    XTaskQueueHandle queue,
+    uint64_t delayInMs
+)
+{
+    auto workPointer = new AsyncWork(std::move(work));
+
```
Comment (jasonsandlin) [inline Source/Common/utils.cpp]: cleanup somewhere?

Comment (jasonsandlin) [reply Source/Common/utils.cpp]: ah, its done in  
   delete context;

---
In reply to: [294009762](https://github.com/microsoft/libHttpClient/pull/434#discussion_r294009762) [](ancestors = 294009762)

File: Source/Common/utils.cpp [hunk 2]
```diff
@@ -76,6 +76,47 @@ bool StringToUint(String const& s, uint64_t& v, int32_t base)
...
+        asyncOp,
+        reinterpret_cast<void*>(workPointer),
+        nullptr,
+        __FUNCTION__,
+        [](XAsyncOp op, const XAsyncProviderData* data)
+        {
+            if (op == XAsyncOp::DoWork)
+            {
+                AsyncWork* work{ reinterpret_cast<AsyncWork*>(data->context) };
+                (*work)();
```
Comment (jasonsandlin) [inline Source/Common/utils.cpp]: i would verify its non-null for safety

File: Source/WebSocket/Websocketpp/websocketpp_websocket.cpp [hunk 3]
```diff
@@ -640,59 +668,83 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
...
-    template <typename WebsocketConfigType>
-    void shutdown_wspp_impl()
+    void send_ping()
     {
-        auto httpSingleton = get_http_singleton(false);
-        if (nullptr == httpSingleton)
-            return;
+        // By design, wspp doesn't raise any sort of event when the client's connection
+        // is terminated (i.e. by disconnecting the network cable). Sending periodic ping
+        // allows us to detect this situation.
```
Comment (rakhad) [inline Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: nit: link to proof

**Pull request merged**